### DISCUSSION TITLE
Getting URL path in span name with Tomcat

### DISCUSSION THREAD

**Discussion by dolzenko** [open-telemetry/opentelemetry-java-instrumentation#8885] [Q&A] [answered]
I want to get path included in span name, or added as attribute. Currently all spans come as 'GET /*'. I assume that might be due to using JRuby RackFilter as below, so that all requests go into my Rails app and routing is done there?

```
<web-app>
  <filter>
    <filter-name>RackFilter</filter-name>
    <filter-class>org.jruby.rack.RackFilter</filter-class>
  </filter>
  <filter-mapping>
    <filter-name>RackFilter</filter-name>
    <url-pattern>/*</url-pattern>
  </filter-mapping>
</web-app>
```

**Comment by trask** [comment, marked as answer]
hi @dolzenko, the Java agent has instrumentation for many common web frameworks to capture their "routes", check out https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks.

unfortunately, there's nothing for rails on jruby

probably your easiest option would be to update the span names in code using Span.current().updateName(...)
